tools/buildsys/src/builder.rs

```diff
@@ -372,19 +385,20 @@ fn create_build_dir(kind: &BuildType, name: &str, arch: &str) -> Result<PathBuf>
 const MARKER_EXTENSION: &str = ".buildsys_marker";
 
 /// Copy build artifacts to the output directory.
-/// Currently we expect a "flat" structure where all files are in the same directory.
 /// Before we copy each file, we create a corresponding marker file to record its existence.
 fn copy_build_files<P>(build_dir: P, output_dir: P) -> Result<()>
 where
     P: AsRef<Path>,
 {
     fn is_artifact(entry: &DirEntry) -> bool {
-        entry.file_type().is_file()
-            && entry
-                .file_name()
-                .to_str()
-                .map(|s| !s.ends_with(MARKER_EXTENSION))
-                .unwrap_or(false)
+        entry.path().is_dir()
+            || (entry.file_type().is_file()
+                && entry
+                    .file_name()
+                    .to_str()
+                    .map(|s| !s.ends_with(MARKER_EXTENSION))
+                    .unwrap_or(false))
+            || (entry.file_type().is_symlink())
```

Comment:
This is getting hard to read


 elsewhere it looks like you have an `is_marker` function,  so maybe something like:

```suggestion
entry.path().is_dir() || is_marker(entry) || entry.file_type().is_symlink()
```

Or if not, you can always store a few booleans along the way:

```rust
let is_dir = entry.path().is_dir();
let is_file = entry.file_type().is_file()
let is_marker = is_file && entry
                                .file_name()
                                .to_str()
                                .map(|s| !s.ends_with(MARKER_EXTENSION))
                                .unwrap_or(false);
is_symlink = entry.file_type().is_symlink();
```

Now the logic is easy to read
```rust
is_dir || is_marker || is_symlink
```